lib/docker_manager/upgrader.rb

```diff
@@ -84,13 +84,22 @@ def upgrade
         run "cd #{repo.path} && git branch -u origin/main main"
         run("cd #{repo.path} && git reset --hard HEAD@{upstream}")
       else
-        run("cd #{repo.path} && git fetch --tags --force && git reset --hard HEAD@{upstream}")
+        run("cd #{repo.path} && git fetch --tags --prune-tags --prune --force")
+
+        if repo.detached_head?
+          run("cd #{repo.path} && git reset --hard")
+          run("cd #{repo.path} && git -c advice.detachedHead=false checkout #{repo.tracking_ref}")
+        else
+          run("cd #{repo.path} && git reset --hard HEAD@{upstream}")
+        end
       end
 
       percent(20 * (index + 1) / @repos.size)
     end
 
-    run("bundle install --deployment --jobs 4 --without test development")
+    run("bundle config --local deployment true")
+    run("bundle config --local without 'development test'")
```

Comment:
Those settings should probably be moved to the docker image itself

Reply:
They are already part of the docker image. I'll try to remove it from the upgrader.